### PR TITLE
[PAPI-389] Use new node endpoint to determine InterFlux token trust

### PR DESCRIPTION
**Pull request by drmathias**

### CODE REVIEW

File: src/Opdex.Platform.Common/Models/Address.cs [hunk 1]
```diff
@@ -56,7 +56,7 @@ public Address(string value)
 
     public override int GetHashCode()
     {
-        return Value.GetHashCode();
+        return Value is null ? "".GetHashCode() : Value.GetHashCode();
```
Comment (mrtpain) [inline src/Opdex.Platform.Common/Models/Address.cs:59]: Doesn't make a difference but could probably use `string.Empty.GetHashCode();`

File: src/Opdex.Platform.Application/PlatformApplicationServiceCollectionExtensions.cs [hunk 2]
```diff
@@ -684,4 +687,10 @@ private static IServiceCollection AddAssemblers(this IServiceCollection services
 
         return services;
     }
+
+    private static IServiceCollection AddCacheServices(this IServiceCollection services)
```
Comment (mrtpain) [inline src/Opdex.Platform.Application/PlatformApplicationServiceCollectionExtensions.cs:691]: Curious about naming, "Cache" doesn't seem very fitting here right?

File: src/Opdex.Platform.Application/Cache/WrappedTokenTrustValidator.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+    public async ValueTask<bool> Validate(Address token, CancellationToken cancellationToken = default)
+    {
+        if (_trustedTokens is null)
+        {
+            var supportedTokens = await _supportedContractsModule.GetList(cancellationToken);
+            _trustedTokens = supportedTokens.Select(t => t.Src20Address).ToImmutableHashSet();
+        }
+
+        return _trustedTokens.Contains(token);
```
Comment (mrtpain) [inline src/Opdex.Platform.Application/Cache/WrappedTokenTrustValidator.cs:33]: Don't think _trustedTokens would ever _not_ be null right?

File: src/Opdex.Platform.Common/Models/Address.cs (hunk 1 above)
Comment (drmathias) [reply src/Opdex.Platform.Common/Models/Address.cs]: Updated

File: src/Opdex.Platform.Application/PlatformApplicationServiceCollectionExtensions.cs (hunk 2 above)
Comment (mrtpain) [reply src/Opdex.Platform.Application/PlatformApplicationServiceCollectionExtensions.cs:691]: That's going to require a redeployment or restart of the API when we deploy a new FN version. 

Thinking now that the list is retrieved from the FN, we add a `Trusted` DB column to the `token_wrapped` table and set it once during token creation. Then we don't need to cache anything and the list would only be retrieved when adding a new token.

**Review by mrtpain — APPROVED** [review]
lgtm, nice work as always

**Comment by drmathias** [comment]
Will hold off from merging until we're able to update the Cirrus node

**Comment by mrtpain** [comment]
@drmathias I added this commit https://github.com/Opdex/opdex-v1-api/pull/223/commits/b7590f1fca6d8e2295a0eafc52facfdbc74bdef9 to adjust the Interflux supported token model to match Stratis FN, also added a temporary static list so we can create liquidity pools while waiting for Stratis.Bitcoin package updates.

**Pull request merged**